code/game/objects/items.dm

```diff
@@ -875,8 +875,11 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 		if(H.mind.get_skill_level(/datum/skill/mining) >= SKILL_LEVEL_JOURNEYMAN && prob(H.mind.get_skill_modifier(/datum/skill/mining, SKILL_PROBS_MODIFIER))) // we check if the skill level is greater than Journeyman and then we check for the probality for that specific level.
 			mineral_scan_pulse(get_turf(H), SKILL_LEVEL_JOURNEYMAN - 2) //SKILL_LEVEL_JOURNEYMAN = 3 So to get range of 1+ we have to subtract 2 from it,.
 
-	delay *= toolspeed * skill_modifier
+	if(tool_behaviour == TOOL_MINING && iscyborg(user))
+		var/mob/living/silicon/H = user
+		skill_modifier = H.mind.get_skill_modifier(/datum/skill/mining, SKILL_SPEED_MODIFIER)
 
+	delay *= toolspeed * skill_modifier
```

Comment:
If I'm not mistaken it works fine without either checks but it's been a long time since I tested so I'm not too inclined to trust remembering that properly, if it does work I have to wonder why the check is even there in the first place

Reply:
Upon further investigation it's even more odd to me as the special actions some could argue shouldn't give exp such as simplemobs smashing rocks... They already don't give exp because give_exp defaults to false, this is literally here to prevent borgs from getting mining exp it seems, unless I missed something

Reply:
Or rather this one is to prevent any non-human mobs from getting the bonuses, the exp is given elsewhere